src/Chapter3.hs

```diff
@@ -971,6 +1146,39 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+days = ["Sun", "Mon","Tue" ,"Wed", "Thu", "Fri", "Sat"]
+
+data WeekDay = WeekDay
+  {
+    day :: String
+  }
```

Comment:
There is a limited amount of weekdays, so it is not needed to be extended, or flexible. Sum type better represents that.

Also, with such representation as you have now, you would need to produce unsafe functions as below 🙂